### PR TITLE
make: replace flake8 with ruff

### PR DESCRIPTION
**Pull request by rst0git**
This pull request is rebased on https://github.com/checkpoint-restore/criu/pull/2333 and https://github.com/checkpoint-restore/criu/pull/2345.

[Ruff](https://github.com/astral-sh/ruff) is a Python linter written in Rust, designed to replace flake8. It is significantly faster and actively maintained.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -96,7 +96,7 @@ Here are some bad examples of clang-format-ing:
 ```
 @@ -58,8 +59,7 @@ static int register_membarriers(void)
          }
- 
+
```
Comment (Snorch) [inline CONTRIBUTING.md:99]: This " " space at the end of line is intended to be there as it is a part of diff format.

File: scripts/build/Dockerfile.alpine [hunk 2]
```diff
@@ -41,13 +41,12 @@ RUN apk add \
...
 	py-yaml \
-	py3-flake8 \
 	py3-importlib-metadata \
 	asciidoctor
 
 # The rpc test cases are running as user #1000, let's add the user
 RUN adduser -u 1000 -D test
 
-RUN pip3 install junit_xml --break-system-packages
+RUN pip3 install junit_xml ruff --break-system-packages
```
Comment (Snorch) [inline scripts/build/Dockerfile.alpine:50]: Why do we need ruff here? (Do we call lint somewhere here?)

File: scripts/ci/prepare-for-fedora-rawhide.sh [hunk 3]
```diff
@@ -23,13 +23,13 @@ dnf install -y \
...
 	protobuf-c-devel \
 	protobuf-devel \
-	python3-flake8 \
 	python3-PyYAML \
 	python3-protobuf \
 	python3-junit_xml \
 	python3-pip \
 	python3-importlib-metadata \
 	python-unversioned-command \
+	ruff \
```
Comment (Snorch) [inline scripts/ci/prepare-for-fedora-rawhide.sh:32]: Same. Do we need ruff here?

**Comment by Snorch** [comment]
I have no objections to switching flake8 -> ruff as far as it works =) It claims `Drop-in parity with Flake8` so it should not be worse.

**Comment by rst0git** [comment]
@Snorch Thank you for the code review! Indeed, it seems that we only need to install Ruff for the lint workflow. I have updated the pull request.

**Pull request merged**